### PR TITLE
Add render option to BlockChain<T>() constructor

### PR DESCRIPTION
**Pull request by earlbread**
This adds `allowRender` option to `BlockChain<T>`.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ To be released.
 
  -  Added `TurnClient.BindProxies()` method. [[#756], [#868]]
  -  Added `ActionEvaluation.Exception` property.  [[#860], [[#875]]]
+ -  Added `InvalidTxGenesisHashException` class.  [[#796], [#878]]
```
Comment (moreal) [inline CHANGES.md:22]: Was it also applied to *9c-beta*?

File: CHANGES.md [hunk 2]
```diff
@@ -39,6 +41,9 @@ To be released.
 [#860]: https://github.com/planetarium/libplanet/issues/860
 [#868]: https://github.com/planetarium/libplanet/pull/868
 [#875]: https://github.com/planetarium/libplanet/pull/875
+[#878]: https://github.com/planetarium/libplanet/pull/878
```
Comment (moreal) [inline CHANGES.md:44]: If #878 was not applied to *9c-beta*, it seems better to remove.

File: CHANGES.md (hunk 1 above)
Comment (earlbread) [reply CHANGES.md]: You're right. Removed it.

File: Libplanet/Blockchain/BlockChain.cs [hunk 3]
```diff
@@ -69,30 +74,40 @@ public class BlockChain<T>
...
+        /// <see cref="BlockChain{T}"/>.</param>
         /// <exception cref="InvalidGenesisBlockException">Thrown when the <paramref name="store"/>
         /// has a genesis block and it does not match to what the network expects
         /// (i.e., <paramref name="genesisBlock"/>).</exception>
         public BlockChain(
             IBlockPolicy<T> policy,
             IStore store,
-            Block<T> genesisBlock
+            Block<T> genesisBlock,
+            bool allowRender = true
```
Comment (dahlia) [inline Libplanet/Blockchain/BlockChain.cs:86]: How about renaming it to `render` which is simpler?

File: Libplanet/Blockchain/BlockChain.cs [hunk 4]
```diff
@@ -69,30 +74,40 @@ public class BlockChain<T>
         /// it checks if the existing genesis block and this argument is the same.
         /// If the <paramref name="store"/> has no genesis block yet this argument will
         /// be used for that.</param>
+        /// <param name="allowRender">Defines whether to allow rendering to be performed on this
+        /// <see cref="BlockChain{T}"/>.</param>
```
Comment (dahlia) [inline Libplanet/Blockchain/BlockChain.cs:78]:
```suggestion
        /// <see cref="BlockChain{T}"/>.  Turned on by default.</param>
```

File: CHANGES.md [hunk 5]
```diff
@@ -19,6 +19,7 @@ To be released.
 
  -  Added `TurnClient.BindProxies()` method. [[#756], [#868]]
  -  Added `ActionEvaluation.Exception` property.  [[#860], [[#875]]]
+ -  Added `allowRender` parameter to `BlockChain<T>()` constructor.  [[#883]]
```
Comment (dahlia) [inline CHANGES.md:22]:
```suggestion
 -  Added `allowRender` optional parameter to `BlockChain<T>()` constructor.  [[#883]]
```

File: Libplanet/Blockchain/BlockChain.cs (hunk 3 above)
Comment (earlbread) [reply Libplanet/Blockchain/BlockChain.cs]: Renamed the parameter to `render`.

File: Libplanet/Blockchain/BlockChain.cs [hunk 6]
```diff
@@ -69,30 +74,40 @@ public class BlockChain<T>
         /// it checks if the existing genesis block and this argument is the same.
         /// If the <paramref name="store"/> has no genesis block yet this argument will
         /// be used for that.</param>
+        /// <param name="render">Defines whether to allow rendering to be performed on this
```
Comment (dahlia) [inline Libplanet/Blockchain/BlockChain.cs:77]:
```suggestion
        /// <param name="render">Defines whether to render actions on this
```

File: Libplanet.Tests/Blockchain/BlockChainTest.cs [hunk 7]
```diff
@@ -330,6 +330,56 @@ public async void ProcessActions()
             Assert.NotNull(state);
         }
 
+        [Fact]
+        public void AllowRender()
```
Comment (dahlia) [inline Libplanet.Tests/Blockchain/BlockChainTest.cs:334]:
```suggestion
        public void Render()
```

File: Libplanet/Blockchain/BlockChain.cs [hunk 8]
```diff
@@ -40,6 +40,11 @@ public class BlockChain<T>
         private readonly object _txLock;
         private readonly ILogger _logger;
 
+        /// <summary>
+        /// Defines whether to allow rendering to be performed on this <see cref="BlockChain{T}"/>.
```
Comment (dahlia) [inline Libplanet/Blockchain/BlockChain.cs:44]:
```suggestion
        /// Defines whether to render actions on this <see cref="BlockChain{T}"/>.
```

File: CHANGES.md [hunk 9]
```diff
@@ -19,6 +19,7 @@ To be released.
 
  -  Added `TurnClient.BindProxies()` method. [[#756], [#868]]
  -  Added `ActionEvaluation.Exception` property.  [[#860], [[#875]]]
+ -  Added `render` optional parameter to `BlockChain<T>()` constructor.  [[#883]]
```
Comment (longfin) [inline CHANGES.md:22]: `BlockChain<T>()` and `BlockChain<T>(render=true)` are different method in IL. so users should recompile their binary with `BlockChain<T>(render=true)` once again. should we mention it? @planetarium/libplanet

Comment (earlbread) [reply CHANGES.md:22]: Is it better we move it to Backward-incompatible API changes?

Comment (earlbread) [reply CHANGES.md]: Moved the changelog to Backward-incompatible API changes.

File: Libplanet/Blockchain/BlockChain.cs [hunk 10]
```diff
@@ -102,7 +117,8 @@ Block<T> genesisBlock
             IStore store,
             Guid id,
             Block<T> genesisBlock,
-            bool inFork
+            bool inFork,
+            bool render = true
```
Comment (dahlia) [inline Libplanet/Blockchain/BlockChain.cs:121]: Since it's just a `private` method, making all parameters required might be better IMHO.

Comment (earlbread) [reply Libplanet/Blockchain/BlockChain.cs]: I made `render` as a required parameter.

**Pull request merged**